lib/big_brother/health_fetcher.rb

```diff
@@ -17,28 +17,31 @@ def self.interpol_status(interpol_nodes, fwmark)
       end
 
       response = _first_interpol_response(urls)
-      response.response_header.status == 200 ? JSON.parse(response.response) : []
-    rescue JSON::ParserError
-      []
+      if response && response.response_header.status == 200
+        begin
+          JSON.parse(response.response)
+        rescue JSON::ParserError
+          []
+        end
+      else
+        []
+      end
     end
 
     def self._first_interpol_response(urls)
       result = nil
       EM::Synchrony::Iterator.new(urls, urls.size).each do |url, iter|
         BigBrother.logger.debug("Fetching health from #{url}")
         http = EventMachine::HttpRequest.new(url, :connect_timeout => 2, :inactivity_timeout => 2).aget
-        this = self
         http.callback do
-          result = http
           if http.response_header.status == 200
             BigBrother.logger.debug("Request to #{url} was successful")
-            this.instance_variable_set(:@ended, true) #This halts the loop
```

Comment:
is halting the loop not necessary? Can you verify that a slower response will not entirely block? we really don't care about all the responses the first 200 response should be adequate, so we should break out of the loop ASAP, and monkeying with the internal state of the iterator seemed like the only way to break out of the loop

Reply:
At least in the scenario of tests, the existence of this line didn't have any bearing on the timings and didn't seen to be breaking out of the loop.

Reply:
I went ahead and put this back just in case I'm wrong, no need to rock that boat.